### PR TITLE
fix: Infinite loading on admin approved list

### LINKED ISSUE #2672: Infinite loading on test-proposals.
https://test-proposals.decred.org/?tab=approved

This tab get caught in an infinite loading loop.

### PR DESCRIPTION
**Pull request by victorgcramos**
Closes #2672.

- Fixes the admin billing changes infinite loading after login with a
previously loaded approved proposals list.

- Adds loading indicator for billing status changes

### Ui Changes
2672-loading-billing

### CODE REVIEW

**Review by lukebp — CHANGES_REQUESTED** [review]
1. Don't add a separate loading animation for the button.  Having multiple loading animations all over the place is distracting and bad UX.  The main loading animation should be displayed until all of the data has been retrieved to display the list.

2. This is caught in an infinte loop on refresh, so something is off anyway.


https://user-images.githubusercontent.com/9039877/143280261-4010f80c-0364-4c86-99da-4f856de169c2.mp4

**Comment by lukebp** [comment]
The infinite loop is actually a loop that fetches all existing approved proposals.  I wasn't able to reproduce it initially becuase I only had a single page of approved proposals in my test env.

**Comment by lukebp** [comment]
Reproduction steps for button infinite loading on firefox:
1. Load Approved tab without being logged in
2. Log in as an admin
3. Navigate to the Approved tab
4. Refresh the page while on the Approved tab

**Review by lukebp — CHANGES_REQUESTED** [review]
@victorgcramos I'm not seeing the infinite loading anymore, but the billing status buttons still disappear on refresh.  The infinite loading was caused by the billing statuses not being fetched.  Looks like this is the same root cause.


https://user-images.githubusercontent.com/9039877/143934901-70dc47e7-0fbd-436f-8dcb-08f76ed7b9df.mp4

**Comment by victorgcramos** [comment]
@lukebp this is how it looks like now. Thanks for the review and the helpful thoughts.

2672-5-approved

**Review by lukebp — APPROVED** [review]
tACK

Needs code approval from @tiagoalvesdulce.

File: src/components/Proposal/ProposalActions.jsx [hunk 1]
```diff
@@ -88,11 +88,10 @@ const PublicActions = ({
...
   const isApproved = isApprovedProposal(proposal, voteSummary);
   const { numbillingstatuschanges } = billingStatusChangeMetadata || {};
+  const needsBillingStatus =
+    !isLegacy && !isRfp && isApproved && currentUser?.isadmin;
   const isSetBillingStatusAllowed =
-    !isLegacy &&
-    !isRfp &&
-    isApproved &&
-    numbillingstatuschanges < billingstatuschangesmax;
+    needsBillingStatus && numbillingstatuschanges < billingstatuschangesmax;
```
Comment (tiagoalvesdulce) [inline src/components/Proposal/ProposalActions.jsx:94]: 👍

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM

**Pull request merged**